espnet2/fileio/sound_scp.py

```diff
@@ -24,7 +24,11 @@ def soundfile_read(
     prev_wav = None
     for wav in wavs:
         with soundfile.SoundFile(wav) as f:
-            array = f.read(dtype=dtype, always_2d=always_2d)
+            # for supporting half-precision training
+            if dtype == "float16":
+                array = f.read(dtype="float32", always_2d=always_2d).astype(dtype)
```

Comment:
Should we touch the IO part to support half-precision training?  Is it possible to do the cast somewhere else(eg. network pre-processor or data loader)?

Reply:
If this line of code is not added and we set `--train_dtype float16`, it will raise an error:
```python
ValueError: dtype must be one of ['float32', 'float64', 'int16', 'int32'] and not 'float16'
```